client/www/scripts/modules/manager/manager.controllers.js

```diff
@@ -638,20 +678,18 @@ Manager.controller('ManagerMainController', [
 
     function updateHostEnv(host, env) {
       var deferred = $q.defer();
-      var command =  {
-        cmd: 'env-set',
-        env: env
-      };
 
-      if (host) {
-        host.action(command, function(err, res) {
-          if (err) {
-            deferred.reject(err);
+      getPmService(host).then(function(service) {
+        for (var key in env) {
```

Comment:
Instead of `.setEnv()` you could call `.setEnvs()` with this `env` object. Keys that have `null` values will be treated as removals.

https://github.com/strongloop/strong-mesh-models/blob/master/common/models/service.js#L47-L63